File: Design/Insert Delete GetRandom O(1)/solution.py
class RandomizedSet:

    def __init__(self):
        self.arr = []
        self.dict = dict()

    def insert(self, val: int) -> bool:
        if val in self.dict: return False
        self.dict[val] = len(self.arr)
        self.arr.append(val)
        return True

    def remove(self, val: int) -> bool:
        if val not in self.dict: return False
        idx = self.dict[val]
        self.dict[self.arr[-1]] = idx
        self.arr[-1], self.arr[idx] = self.arr[idx], self.arr[-1]
        self.dict.pop(self.arr.pop())
        return True

    def getRandom(self) -> int:
        return self.arr[random.randint(0, len(self.arr) - 1)]


# Your RandomizedSet object will be instantiated and called as such:
# obj = RandomizedSet()
# param_1 = obj.insert(val)
# param_2 = obj.remove(val)
# param_3 = obj.getRandom()

File: Design/Insert Delete GetRandom O(1)/solution.ts
class RandomizedSet {
    arr
    map
    constructor() {
        this.map = new Map()
        this.arr = []
    }

    insert(val: number): boolean {
        if(this.map.has(val)) return false
        this.arr.push(val)
        this.map.set(val, this.arr.length - 1)
        return true
    }

    remove(val: number): boolean {
        if(!this.map.has(val)) return false
        let currentIdx = this.map.get(val)
        
        this.map.set(this.arr[this.arr.length - 1], currentIdx);
        this.arr[currentIdx] = this.arr[this.arr.length - 1]
        this.arr.pop()
        this.map.delete(val)
        
        return true
        
    }

    getRandom(): number {
        let rand = Math.floor(Math.random() * this.arr.length)
        return this.arr[rand]
        
    }
}

/**
 * Your RandomizedSet object will be instantiated and called as such:
 * var obj = new RandomizedSet()
 * var param_1 = obj.insert(val)
 * var param_2 = obj.remove(val)
 * var param_3 = obj.getRandom()
 */